### PR TITLE
feat(evals): judge-eval tool for measuring candidate judges (#871)

### PR DESCRIPTION
**Pull request by allenhutchison**
## Summary

References #871.

Adds the **measurement** half of #871: a tool that runs any candidate LLM-as-judge against the human-labelled calibration set committed in #870 (PR #887), and reports agreement vs. ground truth.

Per the PR-1 / PR-2 split agreed in scoping: **this PR only builds the measurement tool and sanity-checks it.** It does **not** change the harness's runtime judge model. A follow-up PR (based on the numbers this tool produces) will bump `DEFAULT_JUDGE_MODEL` to the winner, and #872 will be needed first if the chosen judge is non-Gemini.

## Changes

- **`evals/lib/judge-eval.mjs`** — pure `evaluateJudgeAgainstCalibration(calibration, judgeFn)`. For each tuple with a non-null `human_label` and a clean `judge_error`, calls `judgeFn(criteria, { userMessage, responseText })`, compares to `human_label`. Returns overall agreement, FP / FN counts, and a per-tuple disagreement list with enough context to inspect why the judge flipped. Per-call judge failures are surfaced as a separate count so a transient 429 mid-run doesn't poison the rate.
- **`evals/calibrate-judge.mjs`** + **`npm run eval:calibrate-judge`** — CLI wrapping the library; `--model=`, `--calibration=<path>`, `--json` (machine-readable summary for diffing runs).
- **`test/evals/judge-eval.test.ts`** — 10 tests covering counting (agreement / FP / FN / accuracy), disagreement labelling, skipping (unlabelled tuples, judge-errored tuples), judge-call-error isolation, and defensive coercion.
- **`evals/README.md`** — usage docs under the existing "Judge calibration" section.

## Sanity-check

Ran against the current judge model to verify the tool reproduces the calibration-labelling headline number:

```
$ npm run eval:calibrate-judge -- --model=gemini-2.5-flash

=== Judge agreement on calibration set ===
Judge model:    gemini-2.5-flash
Tuples total:   90
Evaluated:      90
  unlabelled:   0
  judge-error:  0
  call errors:  0

Agreement:      83/90 (92.2%)
Disagreements:  7
  false +:      1  (judge YES, human NO)
  false −:      6  (judge NO,  human YES)
```

**92.2% exactly matches** the calibration's headline (PR #887). Tool works.

### Bonus signal: judge noise at temp=0

The *specific* disagreeing tuples weren't perfectly stable between the calibration-time judge call and this fresh run:

- `rewrite-section-inplace::1` was FN at calibration time; agreed in this run.
- `sort-recent-projects::3` agreed at calibration time; flipped to FN in this run.

Same overall count (7), two tuples shifted. The current judge has measurable nondeterminism even at `temperature: 0` — worth noting for the #871 decision. A more stable judge would be a real win independent of raw accuracy.

## Testing

- `npm test` — 2936 passing (10 new).
- `npm run build` — clean.
- `npm run format-check` — clean.
- **Live sanity-run** documented above.

### Required

- [x] I have read and agree to the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have read and agree to the [AI Policy](../AI_POLICY.md)
- [x] This PR is linked to an approved issue where the approach was discussed with a maintainer
- [x] All CI checks pass (`npm test`, `npm run build`, `npm run format-check`)
- [x] I have tested this change on Desktop (unit suite + live sanity-run — see "Sanity-check")
- [ ] I have verified this change does not break Mobile — N/A, the eval harness is a desktop-only Node tool
- [x] Documentation has been updated (if applicable)
- [x] I understand that I must address all review comments from CodeRabbit and maintainers, or this PR may be closed

### AI-Generated Code

- [x] This PR includes AI-generated or AI-assisted code
- [x] AI tool(s) used: Claude Code
- [x] I have reviewed and understand all AI-generated code in this PR

## Summary by CodeRabbit

* **New Features**
  * Added a CLI workflow and npm command to evaluate an LLM judge against a human-calibrated dataset, with model override and machine-readable JSON output.

* **Documentation**
  * New guide describing how to run judge calibration, available invocation options, reported metrics (agreement, confusion matrix) and per-case disagreement details.

* **Tests**
  * Added comprehensive tests covering counting, skips, error handling, edge cases, and disagreement reporting.

[Review Change Stack](https://app.coderabbit.ai/change-stack/allenhutchison/obsidian-gemini/pull/889?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: evals/lib/judge-eval.mjs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * Judge call failures are surfaced as a separate count rather than failing
+ * the run — a 429 mid-sweep should not poison the agreement rate. The caller
+ * decides whether to retry or accept partial coverage.
+ *
+ * @param {{tuples: Array<object>}} calibration - Parsed `judge-calibration.json`.
+ * @param {(criterion: string, ctx: {userMessage: string, responseText: string}) => Promise<boolean>} judgeFn
+ * @returns {Promise<JudgeEvalResult>}
+ */
+export async function evaluateJudgeAgainstCalibration(calibration, judgeFn) {
+	const tuples = calibration?.tuples || [];
```
[the remark this answers is not in the document]
Comment (allenhutchison) [reply evals/lib/judge-eval.mjs]: Fixed in 91a5d17ae7603cbf4b6157bda15bc21f3fd01d6b — switched to `Array.isArray` and added a regression test covering both `tuples: 'string'` and `tuples: {}` shapes.
